### DISCUSSION TITLE
Document Python Script (not module)

### DISCUSSION THREAD

**Discussion by JamesFraserLogan** [mkdocstrings/mkdocstrings#591] [Q&A] [answered]
Hello,

Thanks very much for making such an awesome package! I was wondering if there was a way to get documentation to auto generate when the python file in question is a python script and not a module.

I have no problem accessing modules and objects, but when I point to a python file that's a script (it has the if name == "__main__" ....) it won't document it.

I access it like this:

::: package.script

And if the file is not a script (i.e. it doesn't have the if name is main part) it documents 100%. If not, I am not getting documentation to generate.

I have searched around and I can't seem to find an answer.

I am using python 3.11.4

Thanks very much!

**Comment by pawamoy** [comment]
Hello! The presence/absence of `if __name__ == "__main__"` in a Python file has no incidence on the collection of data.
The only scenario where a Python file isn't loaded is when it's in a subfolder of a proper package, with an `__init__.py` module missing in this folder:

```
./
    src/
        package/
            __init__.py
            subfolder/
                script.py
```

In the above example, `script.py` is not loaded because Griffe considers it non-importable (and therefore it shouldn't be part of any API docs).

You can confirm by running MkDocs with the verbose option: `mkdocs build -v`. You should see some debug logs saying "missing `__init__.py`?".

**Reply by JamesFraserLogan** [reply, marked as answer]
Hello and thanks very much for your reply, I very much appreciate it.

Your example makes a lot of sense to me, and I personally wouldn't have a module that also has an "entrypoint script" but I am documenting another's code. I am not missing an init.py, it's just a really poor structure of having a package, a package inside that package and then entrypoint scripts there. 

I really appreciate the explanation, it makes perfect sense to me;  thanks!

I found a way to override it, but it's not automatic.

If I specify the "members" in the markdown it just overrides the default, sane behaviour of not having entry points all over the code base.

::: module.submodule.script
       handler: python
       options:
         members:
           - first_function
           - _another_function
           - thirdFunction
Thanks very much again for the reply, and I hope that this discussion helps someone else.

In this case, I am able to get the functions giving their documentation in a very nice fashion.

I really like this project and will let everyone know to use it, how awesome it is, and how quickly my question got answered! Have a great day!

Best Regards,

- James Logan

**Reply by pawamoy** [reply]
I'm not sure to completely understand what the issue was, but if you managed to solve it, that's great!
